src/cursesport.cpp

```diff
@@ -129,103 +128,54 @@ void catacurses::wborder( const window &win_, chtype ls, chtype rs, chtype ts, c
     }
     int i = 0;
     int j = 0;
-    point old = win->cursor; //methods below move the cursor, save the value!
+    point old = win->cursor; // methods below move the cursor, save the value!
 
-    if( ls ) {
-        for( j = 1; j < win->height - 1; j++ ) {
-            mvwaddch( win_, point( 0, j ), ls );
-        }
-    } else {
-        for( j = 1; j < win->height - 1; j++ ) {
-            mvwaddch( win_, point( 0, j ), LINE_XOXO );
-        }
-    }
-
-    if( rs ) {
-        for( j = 1; j < win->height - 1; j++ ) {
-            mvwaddch( win_, point( win->width - 1, j ), rs );
-        }
-    } else {
-        for( j = 1; j < win->height - 1; j++ ) {
-            mvwaddch( win_, point( win->width - 1, j ), LINE_XOXO );
-        }
-    }
-
-    if( ts ) {
-        for( i = 1; i < win->width - 1; i++ ) {
-            mvwaddch( win_, point( i, 0 ), ts );
-        }
-    } else {
-        for( i = 1; i < win->width - 1; i++ ) {
-            mvwaddch( win_, point( i, 0 ), LINE_OXOX );
-        }
-    }
+    const chtype border_ls = ls ? ls : LINE_XOXO;
+    const chtype border_rs = rs ? rs : LINE_XOXO;
+    const chtype border_ts = ts ? ts : LINE_OXOX;
+    const chtype border_bs = bs ? bs : LINE_OXOX;
+    const chtype border_tr = tr ? tr : LINE_OOXX;
+    const chtype border_bl = bl ? bl : LINE_XXOO;
+    const chtype border_br = br ? br : LINE_XOOX;
 
-    if( bs ) {
-        for( i = 1; i < win->width - 1; i++ ) {
-            mvwaddch( win_, point( i, win->height - 1 ), bs );
-        }
-    } else {
-        for( i = 1; i < win->width - 1; i++ ) {
-            mvwaddch( win_, point( i, win->height - 1 ), LINE_OXOX );
-        }
+    for( j = 1; j < win->height - 1; j++ ) {
+        mvwaddch( win_, point( 0, j ), border_ls );
     }
-
-    if( tl ) {
-        mvwaddch( win_, point_zero, tl );
-    } else {
-        mvwaddch( win_, point_zero, LINE_OXXO );
+    for( j = 1; j < win->height - 1; j++ ) {
+        mvwaddch( win_, point( win->width - 1, j ), border_rs );
     }
-
-    if( tr ) {
-        mvwaddch( win_, point( win->width - 1, 0 ), tr );
-    } else {
-        mvwaddch( win_, point( win->width - 1, 0 ), LINE_OOXX );
+    for( i = 1; i < win->width - 1; i++ ) {
+        mvwaddch( win_, point( i, 0 ), border_ts );
     }
-
-    if( bl ) {
-        mvwaddch( win_, point( 0, win->height - 1 ), bl );
-    } else {
-        mvwaddch( win_, point( 0, win->height - 1 ), LINE_XXOO );
+    for( i = 1; i < win->width - 1; i++ ) {
+        mvwaddch( win_, point( i, win->height - 1 ), border_bs );
     }
+    mvwaddch( win_, point_zero, tl ? tl : LINE_OXXO );
```

Comment:
you factored out every other border `chtype` into a constant, why not this one as well for consistency?

Reply:
Missed one. Thanks.